pydantic_numpy/helper/annotation.py

```diff
@@ -17,11 +18,55 @@
 from pydantic_numpy.model import MultiArrayNumpyFile
 
 
+def serialize_numpy_array_to_data_dict(array_like: npt.ArrayLike) -> NumpyDataDict:
+    """
+    Serialize a NumPy array into a data dictionary format suitable for frontend display or processing.
+
+    This function converts a given NumPy array into a dictionary format, which includes the data type
+    and the data itself. If the array contains datetime or timedelta objects, it converts them into integer
+    representations. Otherwise, the array is converted to a floating-point representation. This is particularly
+    useful for preparing NumPy array data for JSON serialization or similar use cases where NumPy's native
+    data types are not directly compatible.
+
+    Note
+    -------
```

Comment:
Are those my dash lines? :rofl:

Can we make them as long as the line above, it hurts to see this :stuck_out_tongue_winking_eye: